mp2-v1/src/values_extraction/mod.rs

```diff
@@ -29,32 +34,77 @@ pub use public_inputs::PublicInputs;
 pub(crate) const KEY_ID_PREFIX: &[u8] = b"KEY";
 pub(crate) const VALUE_ID_PREFIX: &[u8] = b"VAL";
 
+/// Constant prefixes for the inner and outer key IDs of mapping slot.
+/// Restrict to one field.
+pub(crate) const INNER_KEY_ID_PREFIX: &[u8] = b"IN_KEY";
+pub(crate) const OUTER_KEY_ID_PREFIX: &[u8] = b"OUT_KEY";
+
 pub(crate) const BLOCK_ID_DST: &[u8] = b"BLOCK_NUMBER";
 
+/// Storage slot information for generating the proof
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct StorageSlotInfo<const MAX_COLUMNS: usize, const MAX_FIELD_PER_EVM: usize> {
+    slot: StorageSlot,
+    metadata: MetadataGadget<MAX_COLUMNS, MAX_FIELD_PER_EVM>,
+    outer_key_id: F,
+    inner_key_id: F,
+}
+
+impl<const MAX_COLUMNS: usize, const MAX_FIELD_PER_EVM: usize>
+    StorageSlotInfo<MAX_COLUMNS, MAX_FIELD_PER_EVM>
+{
+    pub fn new(
+        slot: StorageSlot,
+        metadata: MetadataGadget<MAX_COLUMNS, MAX_FIELD_PER_EVM>,
+        outer_key_id: F,
```

Comment:
We could take `Option<F>` for `outer_key_id` and `inner_key_id` so that `None` can be provided when this method is called for storage slots where this values are meaningless (e.g., simple variables)? It would avoid the caller to provide dummy values, which might be ambiguous. Wdyt?